canonical/ubuntu-advantage-client
# Version integration test

## Background
We need our tests to print the version under test. We print the version of ua to stdout, but we print _a lot_ of stuff to stdout. So we probably don't want to put all the stdout noise in front of SRU reviewers just for version verification. This is an attempt to verify the version in a way that works with the natural output of behave and works (mostly) like any other integration test.

The way I got the version test to work in such a way that it uses the environment variable and prints everything in a pretty way seems a bit hacky, but wasn't sure of a better way to do it. Let me know if there is a better way.

I also modified how we're skipping tests based  on series which should reduce the amount of noisy output we're currently getting. It also seems a bit hacky... but the output is much nicer

## Proposed Commit Message


> tests: add test to check and output version being tested

> tests: filter on series earlier to avoid extra output 

## Test Steps

- Run `tox -e behave-lxd-20.04` (for example)
- Verify that the version test is the first to run
- Verify that the version test is skipped because you didn't set UACLIENT_BEHAVE_CHECK_VERSION
- Verify that all the different series version tests were skipped and did not displaying the whole scenario before being skipped


- Run `UACLIENT_BEHAVE_CHECK_VERSION=27.1 tox -e behave-lxd-20.04 -- features/_version.feature` (for example)
- Verify that the test fails because the actual version is different

## Desired commit type::


## Checklist:
